psi4/src/psi4/findif/fd_freq_0.cc

```diff
@@ -258,30 +259,25 @@ SharedMatrix fd_freq_0(std::shared_ptr <Molecule> mol, Options &options,
         }
 
         for (int i = 0; i < salcs_pi[h].size(); ++i) {
-            double *v = init_array(3 * Natom);
+            std::shared_ptr<VIBRATION> vib = std::make_shared<VIBRATION>(h, 3 * Natom, evals[i]);
             for (int x = 0; x < 3 * Natom; ++x)
-                v[x] = normal_irr[x][i];
-            VIBRATION *vib = new VIBRATION(h, evals[i], v);
+                vib->lx->set( x, normal_irr[x][i]);
             modes.push_back(vib);
         }
 
-        free(evals);
-        free_block(evects);
-        free_block(normal_irr);
+        //free(evals);
+        //free_block(evects);
+        //free_block(normal_irr);
```

Comment:
We were having real problems with data transport, so I said ok to mem leaks. We'll have to assess lifetime of C-side freq code as to its adherence to good practices.